rep-0144.rst

```diff
@@ -0,0 +1,136 @@
+  make it hard to follow OSes conventions/
+* they must be at least two characters long and must start with an alphabetic character.
+  This rule is simply to force the name of the package to be more human understandable.
+
+Global Rules
+------------
+
+* package names should be specific enough to identify what the package does.
+  For example, a motion planner should not be called ``planner``.
+  If it implements the wavefront propagation algorithm, it might be called
+  ``wavefront_planner``.
+  There's obviously tension between making a name specific and keeping it from becoming
+  overly verbose
+* using catchall names such as ``utils`` should be avoided as they do not scope what goes
+  into the package or what should be outside the package
+* a package name should not contain ``ros`` as it is redundant.
+  Exceptions include core packages and ROS bindings of an upstream library
+  (e.g. ``moveit_ros``)
+* one of ROS's goals is to develop a canonical set of tools for making robots do
+  interesting things.
+  The package name should describe what the package does, not where it came from.
+  Then again, as stated in the rules below, ``if a package is specialized
+  by an entity (lab, company, ...), prepend the name of the entity``.
+  But once the package is commonly used, owned and maintained, that name can be dropped
+  as the package becomes the reference
+* to check whether a name is taken, consult [2]_. If you'd like your
+  repository included in that list, see the tutorial at [3]_
+
+Naming Rules
+------------
+
+The following rules define the different parts of the package name.
+The overall idea is to prepend a name with words that distinguish it from similar 
+functional implementation (e.g.: ``lab``, ``robot``) but in order of importance 
+(e.g.: ``python_robot_lab``).
+Similarly, words that specialize this functionality are appended
+(e.g. ``msgs``, ``config`` ...).
+
+The rules to add those words should be followed in order.
+For prefixes:
+
+* if a package is specialized by an entity (lab, company, ...), prepend the 
+  name of the entity.
+  Once the package is commonly used, owned and maintained, that name can be dropped
+* if a package is specialized for a software project, prepend its name
+* if a package is specialized for a hardware piece, prepend its name
+* if a package is specialized for a robot, prepend its name
+
+For suffixes:
+
+* if a package is a driver, append ``driver``
+* if a package contains any of a ROS message/service/action, append ``msgs``
+* if a package is a plugin for a library, append ``<library_name>_plugins``, e.g. 
+  ``pr2_gazebo_plugins``
+
+Special Suffixes:
+
+* a meta package for a robot should be named ``<name_of_the_robot>_robot``, e.g. 
+  ``pr2_robot``
+* a package containing the URDF and meshes of a robot should be named 
+  ``<name_of_the_robot>_description``, e.g ``pr2_description``
+* if a package is meant for test only, prepend ``test``
```

Comment:
We're under 'special suffixes', so either prepend -> append, or create a new 'special prefixes' section.

I'm not really sure which is the case, because there are lots of `test_xxx`, and `xxx_tests` packages out there. Maybe the former, because you chose `test` in singular?.